src/main/java/uk/gov/companieshouse/model/validator/CompanyNumberValidator.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.companieshouse.model.validator;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.springframework.stereotype.Component;
+
+import uk.gov.companieshouse.config.CompanyNumberConfiguration;
+
+@Component
+public class CompanyNumberValidator implements ConstraintValidator<ValidCompanyNumber, String> {
+
+    private final Pattern prefixListRegex = Pattern.compile("(?i)^([A-Z][A-Z]?)\\b(,[A-Z][A-Z]?)*$");
+
+    private final Pattern companyNumberRegex;
+
+    public CompanyNumberValidator(CompanyNumberConfiguration prefixConfig) {
+
+        String prefixString = prefixConfig.getPrefixes();
+
+        if (!this.prefixListRegex.matcher(prefixString).matches()) {
+            throw new IllegalArgumentException(
```

Comment:
Do we really want to fail here? We could return false (for invalid) and add logging to explain the prefix config is invalid